src/Tasks/XslTransformation.cs

```diff
@@ -174,7 +175,15 @@ public override bool Execute()
             }
             catch (Exception e) when (!ExceptionHandling.IsCriticalException(e))
             {
-                Log.LogErrorWithCodeFromResources("XslTransform.TransformError", e.Message);
+                StringBuilder flattenedMessage = new StringBuilder(e.Message);
+                Exception excep = e;
+                while (excep.InnerException != null)
+                {
+                    excep = excep.InnerException;
+                    flattenedMessage.Append(" ---> ").Append(excep.Message);
+                }
+                Log.LogErrorWithCodeFromResources("XslTransform.TransformError", flattenedMessage.ToString());
+                Log.LogMessage(MessageImportance.Low, e.ToString());
```

Comment:
This seems to be copy-pasted code from https://github.com/dotnet/msbuild/blob/main/src/Tasks/DownloadFile.cs#L131-L139
Let's extract it (e.g. into `TaskLoggingHelper`) and call into that from both locations.

Idealy it would reuse the unwrapping code from [`LogErrorFromException`](https://github.com/dotnet/msbuild/blob/main/src/Shared/TaskLoggingHelper.cs#L892), but add ability to specify the resource string (so adding overload of `LogErrorWithCodeFromResources` that would accept Exception parameter). This way we'd have single code for unwrapping exceptions, influencable via environment variable.
(this was discussed as the optional separate PR for the same issue)

Reply:
Thanks. Have wrapped that in the method GetInnerExceptionMessageString.

Reply:
Thanks!